agentendpoint/patch_beta_windows.go

```diff
@@ -122,7 +127,8 @@ func (r *patchTaskBeta) wuaUpdates(ctx context.Context) error {
 }
 
 func (r *patchTaskBeta) runUpdates(ctx context.Context) error {
-	if err := retryFunc(30*time.Minute, "installing Windows updates", func() error { return r.wuaUpdates(ctx) }); err != nil {
+	// Don't use retry function as wuaUpdates handles it's own retries.
+	if err := r.wuaUpdates(ctx); err != nil {
 		return err
```

Comment:
Wrap error to provide stacktrace?

Reply:
The only code path for the underlying functions is through here so tracing is not a huge concern, and this can return error for a cancelled task and we don't want to wrap that as it can get confusing.